src/server/ghService/import/importPackageUtil.ts

```diff
@@ -1,132 +1,75 @@
-
 import fetch from "node-fetch";
+import { BatchImport, Item } from "../types/import/batchImportModel";
 
-import * as jsYaml from "js-yaml";
-import { ManifestFolderList } from "../types/import/manifestFolderListModel";
-
+import parseYaml from "../helpers/decodingHelper";
 
 const {
   GITHUB_TOKEN,
 } = process.env;
 
-const CONTENTS_BASE_URL = "https://api.github.com/repos/microsoft/winget-pkgs/contents";
-
-//! only call for initial import
-const getManifestFolderPaths = async (): Promise<string[]> => {
-  const manifestFolderList: Promise<ManifestFolderList[]> = await fetch(
-    `${CONTENTS_BASE_URL}/manifests`, {
-      headers: {
-        Authorization: `token ${GITHUB_TOKEN}`,
-      },
-    },
-  ).then((res) => res.json());
-
-  const manifestFolderPaths = await (await manifestFolderList).map(
-    (x) => x.path,
-  );
-
-  return manifestFolderPaths;
-};
-
-const getPackageFolderPaths = async (): Promise<string[]> => {
-  //! only use for inital bulk import
-  const manifestFolderPaths = await getManifestFolderPaths();
-
-  const packageFolders: ManifestFolderList[] = await Promise.all(
-    manifestFolderPaths.map((e) => fetch(`${CONTENTS_BASE_URL}/${e}`, {
-      headers: {
-        Authorization: `token ${GITHUB_TOKEN}`,
-      },
-    }).then((res) => res.json())),
-  );
-
-  const flatPackageFolderPaths: ManifestFolderList[] = packageFolders.flat(
-    packageFolders.length,
-  );
-  const packageFolderPaths = flatPackageFolderPaths.map((x) => x.path);
+const BASE_URL = "https://api.github.com/search/code?q=extension:yaml+repo:microsoft/winget-pkgs+path:/manifests";
+const TAKE = "100";
 
-  return packageFolderPaths;
-};
+let PAGECOUNT: number;
 
-// for people like mongo who cant read the docs
-const handleThreeLevelDeep = async (path: string): Promise<string> => {
-  const downloadUrlPath: ManifestFolderList[] = await fetch(`${CONTENTS_BASE_URL}/${path}`, {
+const getPageCount = async (): Promise<void> => {
+  const model: BatchImport = await fetch(`${BASE_URL}&page=0&per_page=1`, {
     headers: {
       Authorization: `token ${GITHUB_TOKEN}`,
     },
   }).then(res => res.json());
+  const totalCount = model.total_count;
+  const pageCount = Math.ceil(totalCount / parseInt(TAKE, 10));
 
-  const downloadUrl = downloadUrlPath[0].download_url;
-
-  return downloadUrl;
+  PAGECOUNT = pageCount;
 };
 
-const getPackageDownloadUrls = async (): Promise<string[]> => {
-  const packageFolderPaths = await getPackageFolderPaths();
+const getPackgeUrls = async (): Promise<string[]> => {
+  // set page count
+  await getPageCount();
 
-  const downloadUrlPaths: ManifestFolderList[] = await Promise.all(
-    packageFolderPaths.map((path) => fetch(`${CONTENTS_BASE_URL}/${path}`, {
+  let yamlUrls: string[] = [];
+
+  for (let page = 0; page <= PAGECOUNT; page += 1) {
+    // eslint-disable-next-line no-await-in-loop
+    const model: BatchImport = await fetch(`${BASE_URL}&page=${page}&per_page=${TAKE}`, {
```

Comment:
pagination in the github api starts at 1

Reply:
done